### PR TITLE
Support handling symbolic link world folder

### PR DESCRIPTION
**Pull request by Vigilans**
Currently, when the world folder to backup is a symbolic link, `shutil.rmtree` in `remove_worlds` will raise an exception:
```python
# symlinks to directories are forbidden, see bug #1669
raise OSError("Cannot call rmtree on a symbolic link")
```
This PR adds proper support for handling symbolic link by copying/removing related files recursively dereferenced from the symlink to the real folder. 

This PR does not change any behavior in current `QuickBackupM`'s implementation, since the entrypoint to added code is scoped only to symlink files by `while os.path.islink(src_path)`.

### Use case

Suppose `server` folder contains multiple redstone saves downloaded from internet, and MCDR instance uses a symlink folder called `world` / `current` to reference the save we want to start, so we do not need to change `world_names` field in every plugin's config file when switching the world to start.

Here, assume the `current` symlink is pointing to `fortress-farm` folder. With this PR, `QuickBackupM` will copy both `current` symlink and `fortress-farm` folder to the slot, and copying them back when recovering. 

When `current` symlink has been switched to other worlds, behavior in this PR also implies that `current` symlink will switch back to the world the slot saves when recovering, which is intuitive.

### CODE REVIEW

**Comment by Fallen-Breath** [comment]
Please write down the symlink logic in readme so people can know what QBM will/can do

**Comment by Vigilans** [comment]
@Fallen-Breath Documentation updated for symlink world's logic.

File: README.md [hunk 1]
```diff
@@ -173,6 +173,27 @@ mcd_root/
...
+
+    qb_multi/
+        slot1/
+            info.json
+            world -> target_world  # 符号链接复制到了备份槽中
+            target_world/ # 符号链接当前指向的世界一起复制到了备份槽中
+        ...
+```
+
+如果后续符号链接更改了指向的世界的话，执行`!!qb back`会将符号链接恢复至备份槽内保存的世界。
```
Comment (Fallen-Breath) [inline README.md]: 将”符号链接“恢复至“保存的世界”此处是否会有表达不清？毕竟“符号链接”跟“世界存档“是两个概念。是将符号链接文件还原了，还是将符号链接指向的存档文件夹还原了

Comment (Vigilans) [reply README.md]: 是将两者同时还原了。这一段是想说明“符号链接在切换存档后，回档时仍旧能回到备份时用的存档”这一符合直觉的现象，我在新的commit将这一段修改为先介绍`!!qb back`的行为，再说明切换指向的存档后执行回档时的现象。

File: README.md [hunk 2]
```diff
@@ -173,6 +173,27 @@ mcd_root/
...
+
+    qb_multi/
+        slot1/
+            info.json
+            world -> target_world  # 符号链接复制到了备份槽中
+            target_world/ # 符号链接当前指向的世界一起复制到了备份槽中
+        ...
+```
+
+执行`!!qb back`时，会恢复备份槽中保存的世界文件夹，并将符号链接重新设置至备份槽中保存的世界。这表示如果后续该符号链接更改了指向的世界，回档时将恢复到备份时保存的世界
```
Comment (Fallen-Breath) [inline README.md]: 现在的确能说明符号链接被还原了，但没对存档文件夹是否被还原，被如何还原做出所说明，看不出是”将两者同时还原“了

Comment (Vigilans) [reply README.md]: 我在新的commit里将`!!qb back`的行为尽量沿着代码的工作流复述，这样与备份时的说明就是对称的了，并且额外指明了符号链接指向的不同世界是相互独立的，在备份及回档过程中不会互相覆盖，以辅助对“还原”一词的理解。

**Pull request merged**